packages/vsx-registry/src/browser/vsx-extensions-contribution.ts

```diff
@@ -107,4 +156,33 @@ export class VSXExtensionsContribution extends AbstractViewContribution<VSXExten
         }
         return false;
     }
+
+    /**
+     * Installs a local .vsix file after prompting the `Open File` dialog. Resolves to the URI of the file.
+     */
+    protected async installFromVSIX(): Promise<URI | undefined> {
+        const props: OpenFileDialogProps = {
+            title: VSXExtensionsCommands.INSTALL_FROM_VSIX.dialogLabel,
+            openLabel: 'Install',
+            filters: { 'VSIX Extensions (*.vsix)': ['vsix'] },
+            canSelectMany: false
+        };
+        const extensionUri = await this.fileDialogService.showOpenDialog(props);
+        if (extensionUri) {
+            if (extensionUri.path.ext === '.vsix') {
+                const extensionName = this.labelProvider.getName(extensionUri);
+                try {
+                    await this.commandRegistry.executeCommand(VscodeCommands.INSTALL_FROM_VSIX.id, extensionUri);
+                    this.messageService.info(`Completed installing ${extensionName} from VSIX.`);
+                } catch (e) {
+                    this.messageService.error(`Failed to install ${extensionName} from VSIX.`);
+                    console.warn(e);
+                }
+            } else {
+                this.messageService.error('The selected file is not a valid "*.vsix" plugin.');
```

Comment:
If the selected URI is not a `.vsix` file, can we return with `undefined`? Strange to me that this method "randomly" resolves with a URI even if the installation has failed or the selected file is not a valid extension. You can also make the method return type to `Promise<void>`. What do you think?